src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts

```diff
@@ -70,9 +71,9 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
 
   private _data: any;
 
-  @ViewChild('nsmContent') private nsmContent: ElementRef | undefined;
-  @ViewChild('nsmDialog') private nsmDialog: ElementRef | undefined;
-  @ViewChild('nsmOverlay') private nsmOverlay: ElementRef | undefined;
+  @ViewChildren('nsmContent') private nsmContent!: QueryList<ElementRef>;
+  @ViewChildren('nsmDialog') private nsmDialog!: QueryList<ElementRef>;
+  @ViewChildren('nsmOverlay') private nsmOverlay!: QueryList<ElementRef>;
```

Comment:
`@ViewChild` is still correct on Angular 8, you juste have to add the parameter `{ static : false }`. This parameter is also correct in previous versions.

Then you don't have to get the first element of the queryList

Reply:
Yes I have. In order to make it work for all Angular versions, I added a method supported by any Angular version. Since the v8 forces at least two params, I updated it with this universal alternative.

Reply:
Like I said, this second parameter is also supported by any Angular version, he is just optional before v8